nanoFramework.Device.Bluetooth/BluetoothLEDevice.cs

```diff
@@ -315,36 +324,67 @@ internal GattReadResult ReadAttributeValue(ushort AttributeHandle)
             GattCommunicationStatus status = GattCommunicationStatus.Unreachable;
             Buffer buffer = null;
             ushort result = 0;
+            bool securityStarted = false;
+            bool retryRead;
 
             lock (lockObj)
```

Comment:
do you need to lock such a large section? Can't it be possible to lock smaller parts?

Reply:
It's to stop another thread starting a read. Until current one complete or timed out. Also a read can start a pairing operation if it gets an access denied. i.e. result 5 which will become a enum